### PR TITLE
feat(frontend): swap ethers for viem and upgrade wagmi to v1 and all other deps

### PR DESCRIPTION
**Pull request by luzzif**
Needs https://github.com/carrot-kpi/create-carrot-template/pull/3 to be merged

### CODE REVIEW

File: packages/frontend/src/page/index.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Button, ErrorFeedback, Typography } from "@carrot-kpi/ui";
 import { ReactElement, useEffect, useState } from "react";
 import { OraclePage, KPITokenRemotePageProps } from "@carrot-kpi/react";
 import { CollateralData } from "../creation-form/types";
-import { Address, useNetwork, useProvider, useToken } from "wagmi";
+import { Address, useNetwork, usePublicClient, useToken } from "wagmi";
```
Comment (guerrap) [inline packages/frontend/src/page/index.tsx:9]: We can import `Address` as a type.

File: packages/frontend/src/utils/data-decoding.ts [hunk 2]
```diff
@@ -1,40 +1,52 @@
 import { Amount, Fetcher } from "@carrot-kpi/sdk";
-import { Provider } from "@wagmi/core";
-import { BigNumber } from "ethers";
-import { defaultAbiCoder } from "ethers/lib/utils.js";
+import { type PublicClient } from "wagmi";
 import { CollateralData } from "../creation-form/types";
+import { Hex, decodeAbiParameters } from "viem";
```
Comment (guerrap) [inline packages/frontend/src/utils/data-decoding.ts:4]: `Hex` can be imported as type.

File: packages/frontend/src/utils/data-decoding.ts [hunk 3]
```diff
@@ -50,9 +62,9 @@ export const decodeKPITokenData = async (
     return collaterals.some((collateral) => !collateral)
         ? null
         : {
-              collaterals: collaterals as CollateralData[],
+              collaterals: collaterals.slice() as CollateralData[],
```
Comment (guerrap) [inline packages/frontend/src/utils/data-decoding.ts:65]: Why do we need `slice` here? (below too)

File: packages/frontend/src/page/index.tsx (hunk 1 above)
Comment (luzzif) [reply packages/frontend/src/page/index.tsx]: Addressed in 324e4eeffa1b055aabcfbd649a938a64786a3c3b

File: packages/frontend/src/utils/data-decoding.ts (hunk 2 above)
Comment (luzzif) [reply packages/frontend/src/utils/data-decoding.ts]: Addressed in f0c94f69958942158d917d53ea5549df263f0c66

File: packages/frontend/src/utils/data-decoding.ts (hunk 3 above)
Comment (luzzif) [reply packages/frontend/src/utils/data-decoding.ts:65]: We don't actually need this, the cast is enough. The problem is that `viem` returns readonly stuff from `decodeAbiParameters` and the slice was needed to make it writeable basically. Addressed in a392c00c4a1c3caaca5352180d9fedb930fbcc87

**Review by guerrap — APPROVED** [review]

**Pull request merged**